potodo/potodo.py

```diff
@@ -10,153 +10,177 @@
     import polib
     import requests
 except ImportError:
-    # TODO: Have the content of requirements.txt dumped in this error message
     print("You need to install polib and requests to be able to run potodo.")
     sys.exit(1)
 
 from potodo._github import get_reservation_list
+from potodo._po_file import PoFile, get_po_files_from_repo
 
+# TODO: Sort the functions (maybe in different files ?
 
-def get_po_files_from_path(path: str):
+
+def initialize_arguments(above: int, below: int, offline: bool, hide_reserved: bool):
+    """
+    Will initialize the arguments as necessary
     """
-    TODO: Try except if path unvalid
+    if not above:
+        # If above isn't specified, then print all files above 0% (all of them)
+        above = 0
+    if not below:
+        # If below isn't specified, then print all files below 100% (all of them)
+        below = 100
 
-    Will get all po files from given path
+    if above and below:
+        if below < above:
+            # If above and below are specified and that below is superior to above, raise an error
+            raise ValueError("Below must be inferior to above")
 
-    :param path: The path to search `po` files for
+    if not offline and not hide_reserved:
+        # If the reservations are to be displayed, then get them
+        issue_reservations = get_reservation_list()
+    else:
+        # Otherwise, an empty list will do the trick
+        issue_reservations = []
+    return above, below, issue_reservations
 
-    :return: Returns a dict will the folder and the po files in it
-    """
 
-    po_files = [file for file in Path(path).glob("**/*.po") if ".git/" not in str(file)]
+def print_dir_stats(
+    directory_name: str, buffer: list, folder_stats: list, printed_list: list
+):
+    """
+    This function prints the directory name, its stats and the buffer
+    """
+    if True in printed_list:
```

Comment:
```suggestion
    if any(printed_list):
```
Attention c'est une suggestion. J'ai l'impression qu'il n'y a que des booléens dans ta liste mais si ce n'est pas le cas, ma suggestion est mauvaise.

Reply:
Non tu as raison, il n'y à que des booléens dans la liste, mais il faut que je vérifie si TOUS les elements dans cette liste sont faux pour ne pas ecrire ce dossier

Reply:
Si tu veux vérifier qu'ils sont tous faux c'est plutôt `if not any(printed_list)`.

Ta version et celle de awecx sont équivalentes:
```python
from hypothesis import given
from hypothesis.strategies import lists, booleans


@given(lists(booleans()))
def test_awecx(items):
    assert (True in items) is (any(items))

@given(lists(booleans()))
def test_julien(items):
    assert all([item is False for item in items]) is (not any(items))

```

Reply:
Euuuh
```python
>>> l = [True, True]
>>> any(l)
True
>>> l = [True, False]
>>> any(l)
True
```

Reply:
ping @JulienPalard

Reply:
oui.

Reply:
I decided to leave the current form as it works fine.